casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala

```diff
@@ -116,13 +117,19 @@ object InterpreterUtil {
               Log[F].warn(s"Found unknown failure") *> Right(none[StateHash])
                 .leftCast[BlockException]
                 .pure[F]
+            case UnusedCommEvent(_) => throw new RuntimeException("found UnusedCommEvent")
```

Comment:
You have a `Sync` constraint, so why not `raiseError`?

Reply:
👍